modules/facilities_api/app/services/facilities_api/v1/ppms/client.rb

```diff
@@ -81,64 +77,75 @@ def specialties_url
           '/dws/v1.0/Specialties'
         end
 
-        def fetch_lat_long_and_radius(params)
-          latitude = Float(params.values_at(:lat, :latitude).compact.first).round(DEGREES_OF_ACCURACY)
-          longitude = Float(params.values_at(:long, :longitude).compact.first).round(DEGREES_OF_ACCURACY)
-          radius = Integer(params.fetch(:radius)).clamp(RADIUS_MIN, RADIUS_MAX)
+        def facility_service_locator_params(params)
+          base_location_params(params).merge(specialty_codes(params))
+        end
 
-          [latitude, longitude, radius]
+        def provider_locator_params(params)
+          if params[:page].present? && params[:per_page].present?
+            _page, _per_page, max_results = fetch_pagination(params)
+
+            base_location_params(params)
+              .merge({ maxResults: max_results })
+              .merge(specialty_codes(params))
+          else
+            base_params(params).merge(specialty_codes(params))
+          end
         end
 
-        def facility_service_locator_params(params)
-          page = Integer(params[:page] || 1)
-          per_page = Integer(params[:per_page] || PER_PAGE)
+        def pos_locator_params(params, pos_code)
+          base_params(params).merge(posCodes: pos_code)
+        end
 
+        def base_params(params)
+          _page, _per_page, max_results = fetch_pagination(params)
```

Comment:
The pagination wasn't being used before, but since the facility locator has to use it, it's included here. The other endpoints using the base parameters have also used the pagination.